src/vs/base/parts/tree/browser/treeModel.ts

```diff
@@ -227,7 +227,9 @@ export class Item extends Events.EventEmitter {
 		this.element = element;
 
 		this.id = id;
-		this.registry.register(this);
+		if (!this.registry.isRegistered(this.id)) {
+			this.registry.register(this);
+		}
```

Comment:
So that means that item IDs are not unique any more. Is that really the case (looks like they are moved around rather than being copied. btw. do we want this behaviour? If we generate js files in separate directory do we want this directory to be gone? There is no control over it at the moment). Do we want these IDs not to be unique? Isn't it just the matter of referring to the same file in many places (or to move it across)?